cds_ils/importer/views.py

```diff
@@ -0,0 +1,169 @@
+        """Custom exception class to be thrown when error occurs."""
+
+        def __init__(self, message, status, payload=None):
+            self.message = message
+            self.status = status
+            self.payload = payload
+
+    @blueprint.errorhandler(RequestError)
+    def handle_request_error(error):
+        """Catch RequestError exception, serialize into JSON, and respond."""
+        payload = dict(error.payload or ())
+        payload['status'] = error.status
+        payload['message'] = error.message
+        return jsonify(payload), error.status
+
+    @blueprint.route('/importer/check/<log_id>', methods=['GET'])
+    @need_permissions("document-importer")
+    def check(log_id):
+        log = ImporterTaskLog.query.filter_by(id=log_id).first()
+        if log:
+            entries = ImporterTaskEntry.query \
+                .filter_by(import_id=log_id) \
+                .order_by(ImporterTaskEntry.entry_index.asc()) \
+                .all()
+            obj = {
+                "id": log_id,
+                "state": log.status.value,  # enum value
+                "start_time": arrow.get(log.start_time).isoformat(),
+                "end_time": arrow.get(log.end_time).isoformat() \
+                if log.end_time else None,
+                "original_filename": log.original_filename,
+                "provider": log.provider,  # string
+                "mode": log.mode.value,  # enum value
+                "source_type": log.source_type,  # string
+            }
+            if log.entries_count:
+                obj["total_entries"] = log.entries_count
+                obj["loaded_entries"] = len(entries)
+            reports = []
+            for entry in entries:
+                if not entry.error:
+                    reports.append({
+                        "index": entry.entry_index,
+                        "success": True,
+                        "report": {
+                            "ambiguous_documents": entry.ambiguous_documents,
+                            "ambiguous_eitems": entry.ambiguous_eitems,
+                            "created_document": entry.created_document,
+                            "created_eitem": entry.created_eitem,
+                            "updated_document": entry.updated_document,
+                            "updated_eitem": entry.updated_eitem,
+                            "deleted_eitems": entry.deleted_eitems,
+                            "series": entry.series,
+                            "fuzzy_documents": entry.fuzzy_documents,
+                        }
+                    })
+                else:
+                    reports.append({
+                        "index": entry.entry_index,
+                        "success": False,
+                        "message": entry.error,
+                    })
+            obj["reports"] = reports
+            return obj
+        else:
+            raise Exception("Task not found")  # TODO
```

Comment:
```suggestion
            return 404
```

If you want to keep it cleaner, maybe return at the top:
```
if not log:
    return 404
```
or extract:
```
if log:
    return get_report(log)
else:
    return 404
```